src/vendor/github.com/xelabs/go-mysqlstack/proto/statement.go

```diff
@@ -156,87 +156,80 @@ func PackStatementExecute(stmtID uint32, parameters []sqltypes.Value) ([]byte, e
 }
 
 // UnPackStatementExecute -- unpack the stmt-execute packet from client.
-func UnPackStatementExecute(data []byte, paramsCount uint16, parseValueFn func(*common.Buffer, querypb.Type) (interface{}, error)) (*Statement, error) {
+func UnPackStatementExecute(data []byte, prepare *Statement, parseValueFn func(*common.Buffer, querypb.Type) (interface{}, error)) (byte, error) {
```

Comment:
The func returns is cursor type, but i have not seen where to use?
Why we only return error?

Reply:
Yes, It is temporarily useless, but it exists on the procotol.

we should use the stmt.ParamsType,
so before the code, the relationship between protoStmt and stmt is created. 
and delete the stmt.BindVars = protoStmt.BindVars.